staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/sample_and_watermark.go

```diff
@@ -160,17 +160,14 @@ func (saw *sampleAndWaterMarkHistograms) SetX1(x1 float64) {
 }
 
 func (saw *sampleAndWaterMarkHistograms) innerSet(updateXOrX1 func()) {
-	var when time.Time
-	var whenInt int64
-	var acc sampleAndWaterMarkAccumulator
-	var wellOrdered bool
-	func() {
+	when, whenInt, acc, wellOrdered := func() (time.Time, int64, sampleAndWaterMarkAccumulator, bool) {
```

Comment:
At this point it seems like there's not much point in keeping it as an anonymous function, might as well just move that whole block out to `func (saw *sampleAndWaterMarkHistograms) validateQuantization(time.Time, int64, sampleAndWaterMarkAccumulator, bool)` (name it something better) unless the intent is to revert all of this once the golang bug is fixed.

Reply:
the point of the inner func is to unlock via a `defer` as soon as our calculations are done before we log or call out to record metrics

Reply:
The side effects argument makes sense I suppose. I get that it just exists to handle the unlock cleanly, but to me the code picks up a bit of an odor when anonymous functions grow return values and such. I'm just here as peanut gallery though, feel free to ignore my feedback.

Reply:
I agree with @liggitt , the inner function should never be invoked in any other context and so I would prefer to not expose it.